verifier/src/x509.rs

```diff
@@ -6,5 +6,179 @@ mod certs;
+impl TryFrom<String> for UnverifiedCertChain {
+    type Error = Error;
+
+    fn try_from(mut pem: String) -> Result<Self> {
+        // Null terminate for Mbedtls
+        pem.push('\0');
+        let certs = Certificate::from_pem_multiple(pem.as_bytes())?;
+        Ok(Self(certs))
+    }
+}
+
+/// A verified certificate chain.
+///
+/// See [`UnverifiedCertChain::verify`] for creating one.
+pub struct VerifiedCertChain(MbedtlsList<Certificate>);
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use alloc::format;
+
+    const LEAF_CERT: &str = include_str!("../data/tests/leaf_cert.pem");
+    const PROCESSOR_CA: &str = include_str!("../data/tests/processor_ca.pem");
+    const ROOT_CA: &str = include_str!("../data/tests/root_ca.pem");
+
+    #[test]
+    fn cert_chain_from_one_pem_cert() {
+        let cert_chain =
+            UnverifiedCertChain::try_from(LEAF_CERT).expect("failed to parse cert chain");
+        // Counting manually, because `MbedtlsList` is a linked list without
+        // `len()` method.
+        let count = cert_chain.0.iter().fold(0, |sum, _| sum + 1);
+        assert_eq!(count, 1);
+    }
+
+    #[test]
+    fn cert_chain_from_two_pem_certs() {
+        let raw_pems = format!("{LEAF_CERT}\n{PROCESSOR_CA}");
+        let cert_chain =
+            UnverifiedCertChain::try_from(raw_pems.as_str()).expect("failed to parse cert chain");
+        let count = cert_chain.0.iter().fold(0, |sum, _| sum + 1);
+        assert_eq!(count, 2);
+    }
+
+    #[test]
+    fn cert_chain_from_invalid_pem_cert() {
+        assert!(matches!(
+            UnverifiedCertChain::try_from(&LEAF_CERT[1..]),
+            Err(Error::MbedTls(_))
+        ));
+    }
+
+    #[test]
+    fn verify_valid_cert_chain() {
+        let raw_pems = format!("{LEAF_CERT}\n{PROCESSOR_CA}\n{ROOT_CA}");
+        let cert_chain =
+            UnverifiedCertChain::try_from(raw_pems.as_str()).expect("failed to parse cert chain");
+        let trust_anchor = TrustAnchor::try_from(ROOT_CA).expect("failed to parse root cert");
+        assert_eq!(cert_chain.verify(&trust_anchor).is_ok(), true);
+    }
+
+    #[test]
+    fn invalid_cert_chain() {
+        let raw_pems = format!("{LEAF_CERT}\n{ROOT_CA}");
+        let cert_chain =
+            UnverifiedCertChain::try_from(raw_pems.as_str()).expect("failed to parse cert chain");
+        let trust_anchor = TrustAnchor::try_from(ROOT_CA).expect("failed to parse root cert");
+        assert!(matches!(
+            cert_chain.verify(&trust_anchor),
+            Err(Error::MbedTls(_))
+        ));
+    }
+
+    #[test]
+    fn unordered_cert_chain_succeeds() {
+        let raw_pems = format!("{PROCESSOR_CA}\n{ROOT_CA}\n{LEAF_CERT}");
+        let cert_chain =
+            UnverifiedCertChain::try_from(raw_pems.as_str()).expect("failed to parse cert chain");
+        let trust_anchor = TrustAnchor::try_from(ROOT_CA).expect("failed to parse root cert");
+        assert_eq!(cert_chain.verify(&trust_anchor).is_ok(), true);
```

Comment:
Any reason for assert_eq?

```suggestion
        assert!(cert_chain.verify(&trust_anchor).is_ok());
```